### PR TITLE
apps/ should default to using libHalide.so rather than libHalide.a

### PR DESCRIPTION
**Pull request by steven-johnson**
This is arguably a better choice.

(My ulterior motivation is to make adding wasm benchmarking to several of the apps, and compiling WITH_V8 is currently much easier to manage link-time dependencies of when using libHalide.so)

### CODE REVIEW

**Comment by abadams** [comment]
I keep thinking we should make this change, but didn't realize it was so simple. The binaries are absurdly large currently.

**Comment by steven-johnson** [comment]
The mingw failure is odd; one of the apps is failing to load libHalide.dll, but the others aren't. I've just tried on the buildbot that's failing and can't replicate. Going to retry the build.

**Comment by steven-johnson** [comment]
My failure-to-repeat before is now easy-to-repeat after trying again. stay tuned.

**Comment by steven-johnson** [comment]
TL;DR: Windows DLL search order is a nightmare; our existing tests happen to work because they are all in the same `bin/` dir as libHalide.dll, but this isn't the case for `apps/`.
